### DISCUSSION TITLE
CreateWindowEx Always return null

### DISCUSSION THREAD

**Discussion by LigaLiao** [microsoft/WindowsAppSDK#1715] [Q&A] [answered]
I try to create transparent windows in the traditional way, 
but CreateWindowEx Always return null.
Is this a deliberate restriction?

**Comment by riverar** [comment]
Please provide code that reproduces the issue.

**Reply by LigaLiao** [reply]
![QQ截图20211104230258](https://user-images.githubusercontent.com/12179645/140338564-ab96a12c-3cf3-4216-a3dc-9612a7e8d9b8.png)

**Comment by LigaLiao** [comment]
`HINSTANCE hInstance = GetModuleHandle(NULL);

        HWND hwnd = CreateWindowEx(
            0,                              // Optional window styles.
            L"CreateWindows",                     // Window class
            L"Test",    // Window text
            WS_OVERLAPPEDWINDOW,            // Window style

            // Size and position
            CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT,

            NULL,       // Parent window    
            NULL,       // Menu
            hInstance,  // Instance handle
            NULL        // Additional application data
        );

        ShowWindow(hwnd, SW_SHOWDEFAULT);
        UpdateWindow(hwnd);`

**Reply by castorix** [reply]
What does  GetLastError(); return ?
I create a window after MddBootstrapInitialize without any problem

**Reply by LigaLiao** [reply]
GetLastError return 1407.

**Comment by riverar** [comment]
Recommending posting your question on Microsoft Q&A with tags `C++` and/or `Windows-API` https://docs.microsoft.com/en-us/answers/topics/614766/windows-api.html

This repository is for Windows App SDK specific issues/questions, not general Win32 programming advice/troubleshooting.

**Reply by LigaLiao** [reply]
I'm here to ask questions because I suspect that the Windows App SDK has restriction on using the Win32 API.
Please tell me if there are restrictions. If not, I can figure it out for myself.

**Reply by riverar** [reply]
Try creating a small C++ console app to reproduce the issue. None of the code we've seen so far seems to be using WinAppSdk.

**Comment by LigaLiao** [comment, marked as answer]
I solved it because of the lack of RegisterClass.
I'm like a clown... 
sorry!

**Reply by riverar** [reply]
Glad you got it sorted out!
